### DISCUSSION TITLE
NOS3 and FreeRTOS

### DISCUSSION THREAD

**Discussion by Gonta01** [nasa/nos3#310] [General]
In [NOS3 cFS Development doc](https://github.com/nasa/nos3/wiki/NOS3-cFS-Development) it is mentioned that usage of cFS and it OSAL "...allows the FSW written for the FreeRTOS target to be built to execute on Linux and the opposite remains true. This makes NOS3 an ideal development environment when using the OSAL Linux target."

Does it mean that NOS3 could be potentially used for board that runs freeRTOS (not Linux)?

**Comment by jlucas9** [comment]
Hi @Gonta01,
The NOS3 (really cFS) can easily be configured to support other operating systems thanks to the OSAL layer. I don't believe FreeRTOS is supported out of the box any longer, but RTEMS and others are certainly.
